common/scala/src/main/scala/whisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -135,9 +153,20 @@ class KafkaConsumerConnector(
   }
 
   private def recreateConsumer(): Unit = {
-    val oldConsumer = consumer
-    oldConsumer.close()
-    logging.info(this, s"old consumer closed")
+    logging.info(this, s"recreating consumer for '$topic'")
+    try {
+      val oldConsumer = consumer
```

Comment:
Is this reassignment necessary. Is just calling `consumer.close()` possible instead?

Reply:
Ehr yes actually. The reassignment is a bit of legacy code. The consumer close was wrapped in a Future earlier.

Reply:
Thanks. Makes sense. I just kept the original code unchanged. Changing now.